### PR TITLE
Evm 224 smart contract related tests

### PR DESCRIPTION
**Pull request by rockNicktive**

### CODE REVIEW

**Comment by rockNicktive** [comment]
First iteration done

**Review by akrckn — APPROVED** [review]

File: qa/sc_evm_test_erc20.py [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+        res = smart_contract.static_call(node, method, addr, uint,
+                                         fromAddress=source_addr,
+                                         gasLimit=10000000, gasPrice=10, toAddress=contract_address)
+    else:
+        res = smart_contract.call_function(node, method, addr, uint,
+                                           fromAddress=source_addr,
+                                           gasLimit=10000000, gasPrice=10, toAddress=contract_address)
+    if generate_block:
+        print("generating next block...")
+        generate_next_blocks(node, "first node", 1)
```
Comment (i-Alex) [inline qa/sc_evm_test_erc20.py:47]: just a hint fro future: we have another method to generate a single block: `generate_next_block(node, "first node")`

**Review by i-Alex — APPROVED** [review]
Amazing!

File: qa/sc_evm_test_erc20.py (hunk 1 above)
Comment (rockNicktive) [reply qa/sc_evm_test_erc20.py:47]: I know, afaik that one is used inside `generate_next_blocks` right? I just wanted to keep block generation mostly uniform throughout the scripts

**Pull request merged**